LoopKit/DeviceManager/BolusActivationType.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  BolusActivationType.swift
+//  LoopKit
+//
+//  Created by Nathaniel Hamming on 2023-09-07.
+//  Copyright © 2023 LoopKit Authors. All rights reserved.
+//
+
+public enum BolusActivationType: String, Codable {
+    case manualNoRecommendation
+    case manualRecommendationAccepted
+    case manualRecommendationChanged
+    case automatic
+
+    public var isAutomatic: Bool {
+        self == .automatic
+    }
+
+    static public func activationTypeFor(recommendedAmount: Double?, bolusAmount: Double) -> BolusActivationType {
```

Comment:
nit: this doesn't handle automatic boluses; maybe `manualActivationTypeFor(...)`?

Reply:
hm... there is a case for `.automatic`. Not sure I follow

Reply:
oh... I see. I can update for `.automatic`